### PR TITLE
Add "Getting Started - Software" guide

### PR DESCRIPTION
**Pull request by mciantyre**
- Add a written guide for first-time software installation
- Include a shell script to automate most of that first-time software
  installation (Linux / macOS only)
- Test that shell script

Looking for feedback on the docs quality, making sure that its easy
to read, and that it includes all of the dependencies and tools.

### CODE REVIEW

**Comment by mciantyre** [comment]
Checks for `first-time-setup.sh` [passed in CI](https://github.com/ebelski/rust-copter/pull/35/checks?check_run_id=1766151570). I transitioned the test to a cron job, instead of running them on every PR / commit, since they run slower (need to download toolchains, build tools, build all code). Could use a dependency cache to avoid the crate download, but lets consider that if the cron job doesn't work out.

**Comment by mciantyre** [comment]
Consider this getting started documentation for #23.

**Review by ebelski — CHANGES_REQUESTED** [review]
Under the "Install Rust" header, it says "...Visual Studio 2019. Read the relevant sections for more information". 
-> Are we pointing the user to somewhere in our documentation or the Rust programming documentation or the linker/build tools documentation? Would it be possible to say something like "...Read the relevant sections, in so and so docs, for more information"?

I plan on following these instructions for a clean install at some point, but so far I think you covered everything and in the right order

File: docs/GettingStartedSoftware.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+by following the instructions in [The Rust Programming Language][the-book-ch1]
+book. We assume that your installation will include `rustup` for managing Rust
+toolchains.
+
+[the-book-ch1]: https://doc.rust-lang.org/book/ch01-01-installation.html
+
+Make sure that you have all additional dependencies. If you're on Linux /macOS,
+you might need to install a linker. If you use Windows, you may need to install
+the Build Tools for Visual Studio 2019. Read the relevant sections for more
+information.
```
Comment (mciantyre) [inline docs/GettingStartedSoftware.md]: @ebelski I'll make it read something like

```suggestion
the Build Tools for Visual Studio 2019. Read the relevant sections of
[The Rust Programming Language] (Chapter 1) for more information.
```

**Review by ebelski — APPROVED** [review]
Thanks!

**Pull request merged**